### DISCUSSION TITLE
Add flake 'flake:nix-on-droid-path' to flake registries

### DISCUSSION THREAD

**Discussion by 573** [nix-community/nix-on-droid#326] [Ideas]
Then I could i. e. use `nix profile` commands instead of `nix-env`.

See here as a reference: https://github.com/Gerschtli/nix-config/blob/6b44f1c/files/apps/setup.sh

**Comment by Gerschtli** [comment]
Please elaborate further. This link only points to a shell script.

**Reply by 573** [reply]
l. 85 (`nix-env --set-flag priority 1000 nix`) could be replaced by `nix profile install $flake2Dir --priority 1000` (as added via https://github.com/NixOS/nix/pull/6522/files#diff-7f2662e78dabad4adb86f56ad39f45893e8afe017dd3143168c5d810a125a727) potentially.

**Reply by 573** [reply]
This way I could do stuff like `nix profile install nixpkgs/d1c3fea7ecbed758168787fe4e4a3157e52bc808#haskellPackages.hledger-iadd` (see https://lazamar.co.uk/nix-versions/?channel=nixpkgs-unstable&package=hledger-iadd) without breaking my current `nix-env`-profile based setup.

**Reply by 573** [reply]
The issue here is probably rooted deeper as it dawns on me now.

**Reply by Gerschtli** [reply]
I am confused. What exactly do you want to achieve that is currently not possible? nix-env vs nix profile is just how you locally imperatively install packages. nix-on-droid works with both.

And how does all of that relate to flake registries? Are you taking about the global registry or a local one?

**Reply by 573** [reply]
IC, it's confusing, no I was trying to replace the mentioned line in the setup script as I'm using it.
Background was that after I imperatively installed `hledger-iadd` I could no longer use the script as is.
Then it was all trial and error ending in reinstalling nix-on-droid entirely. The error output pretty much told me what to do `nix profile` and sorts but I couldn't make it work.

What I was trying was `nix profile install flake:nix-on-droid` because I wouldn't know what else to do when I had to remove the entry `0 - - <store-path-to/nix-on-droid-path>` from the `nix profile list` (output) safely. Removing because that entry seemed to have an incompatible (as in `v3` instead of `v5`) version manifest(.nix or .json don't remember exactly) \* 'obviously' as the derivation belonging was installed using `nix-env` vs. `nix profile` (?)

Render me confused 2.

\* this was part of the error message when running the setup.sh linked.

**Reply by 573** [reply]
> flake registry

to enable this `nix profile install flake:nix-on-droid` FWIW

**Reply by Gerschtli** [reply]
You could just use `nix profile install github:nix-community/nix-on-droid` AFAIK.

I still don't get your original problem without the commands you used and their error messages. Please provide more information

**Reply by 573** [reply]
I will try everything again when my nix-on-droid is back up and will then attach the errors / outputs here, maybe I figure out if it works in the fakedroid environment first.

**Comment by 573** [comment]
Tried the `nix profile install github:nix-community/nix-on-droid#nix-on-droid` approach and when following the suggestion in the error message to install prioritizing the existing package as setup.sh also does:

```
$ nix profile install github:nix-community/nix-on-droid#packages.aarch64-linux.nix-on-droid --priority 6
-bash-5.2$ nix profile list
Index:              0
Store paths:        /nix/store/jag6xazfq1x6nnqx1wa0ivyl8w671hnj-nix-on-droid-path

Index:              1
Flake attribute:    packages.aarch64-linux.nix-on-droid
Original flake URL: github:nix-community/nix-on-droid
Locked flake URL:   github:nix-community/nix-on-droid/039379abeee67144d4094d80bbdaf183fb2eabe5
Store paths:        /nix/store/vaxb3v86g5v6qsw8gc94p49c9gkqv87w-nix-on-droid
```

only then realizing it is two different things and I still wouln't know how to get that nix-on-droid-**path** thing transferred from old manifest to profile-manifest (as in nix profile not nix-env).

Would that be `nix profile install path:/path/to/home/.config/nix-on-droid` then ?

**Comment by 573** [comment]
To illustrate the error I got with nix-profile-installed derivations it read:

```
nix-env -q --json
error: profile '/nix/var/nix/profiles/per-user/nix-on-droid/profile' is incompatible with 'nix-env'; please use 'nix profile' instead
```
Reason, why the original setup.sh wouldn't work anymore as soon as I started using `nix profile [updating operations]`.

**Comment by 573** [comment]
manifest[.nix] before `nix profile install`:

1 2 3 4 /n/v/n/p/p/n/profile-5-link     

   bin/
   etc/
   include/                            
   lib/
   libexec/
   sbin/                               
   share/                              
   manifest.nix@
   rplugin.vim@                        


                                        


                                        

1/9 1970-01-01 01:00 lr-xr-xr-x 65B ->/n

**Comment by 573** [comment]
manifest[.json] after `nix profile install`:

1 2 3 4 /n/v/n/p/p/n/profile-6-link

   bin/
   etc/                                
   include/
   lib/
   libexec/                            
   sbin/                               
   share/
   manifest.json
   rplugin.vim@







1/9 1970-01-01 01:00 dr-xr-xr-x 12K

**Comment by 573** [comment]
```nix
[ { meta = { }; name = "nix-on-droid-pat
h"; out = { outPath = "/nix/store/jag6xa
zfq1x6nnqx1wa0ivyl8w671hnj-nix-on-droid-
path"; }; outPath = "/nix/store/jag6xazf
q1x6nnqx1wa0ivyl8w671hnj-nix-on-droid-pa
th"; outputs = [ "out" ]; type = "deriva
tion"; } ]                              
~                                       
~                                       
~                                       
~                                       
~                                       
~                                       
~                                       
~                                       
~                                       
~                                       
manifest.nix [RO]     1,1            All
```
```json
{"elements":[{"active":true,"priority":5
,"storePaths":["/nix/store/jag6xazfq1x6n
nqx1wa0ivyl8w671hnj-nix-on-droid-path"]}
,{"active":true,"attrPath":"packages.aar
ch64-linux.nix-on-droid","originalUrl":"
github:nix-community/nix-on-droid","outp
uts":null,"priority":6,"storePaths":["/n
ix/store/vaxb3v86g5v6qsw8gc94p49c9gkqv87
w-nix-on-droid"],"url":"github:nix-commu
nity/nix-on-droid/039379abeee67144d4094d
80bbdaf183fb2eabe5"}],"version":2}
~
~
~
~
~
~
manifest.json [RO]    1,1            All
```

**Comment by 573** [comment]
Ok, `nix profile diff-closures` shows me, that I can go back safely at this time (only one step that was `nix profile install ...` see further up).

A `nix profile rollback` does bring back `nix-env -q --json | jq ".[].pname"` to me for free.

And also now I see that setup.sh does nothing anyway in the line cited further up, at least on nix-on-droid it doesn't.

So the question that remained would rather be: What is equivalent to `nix-env -q --json | jq ".[].pname"` using `nix profile`.

Answered that I could well replace the by then offending line in setup.sh when eventually  switching to `nix profile`.

**Comment by 573** [comment]
There seems to be no equivalent to `nix-env -q`.
Even `[[ -d $(nix eval --raw '.#nixosConfigurations.DANIELKNB1.pkgs.nixVersions.unstable.outPath') ]]` does not help me here (`nixosConfigurations.DANIELKNB1.` part is too static) and setting the priority of an already imperatively / manually installed package.
